### PR TITLE
fix(typescript): use correct ts version and fix promise types

### PR DESCRIPTION
**Pull request by jeffbcross**
* Updates TS version
 * Makes aot-friendly packaging (including UMD bundle)
 * Runs Karma against bundle
 * Tests that prod build step does not error on Travis

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -44,6 +44,7 @@
   "devDependencies": {
     "@angular/compiler-cli": "^0.5.0",
     "@angular/platform-server": "^2.0.0-rc.5",
+    "@types/es6-promise": "0.0.31",
```
Comment (alexeagle) [inline package.json]: this is bad, why can't you change the other side to use lib.es2015.promise.d.ts instead?

File: package.json [hunk 2]
```diff
@@ -69,7 +70,7 @@
     "traceur": "0.0.96",
     "tsd": "^0.6.5",
     "typedoc": "github:jeffbcross/typedoc",
-    "typescript": "next",
+    "typescript": "^1.9.0-dev.20160627-1.0",
```
Comment (davideast) [inline package.json]: Firebase requires 2.0+.

File: package.json (hunk 1 above)
Comment (jeffbcross) [inline package.json]: lol

File: src/database/query_observable.ts [hunk 3]
```diff
@@ -1,24 +1,24 @@
 import { Observable } from 'rxjs/Observable';
-import { ScalarObservable } from 'rxjs/observable/ScalarObservable';
+import 'rxjs/add/observable/of';
```
Comment (robwormald) [inline src/database/query_observable.ts]: can you use the non-sideeffecty version instead?

File: src/database/query_observable.ts [hunk 4]
```diff
@@ -1,24 +1,24 @@
...
+import {
+  Query,
+  ScalarQuery,
+  OrderByOptions,
+  OrderBySelection,
+  LimitToOptions,
+  LimitToSelection,
+  Primitive
 } from '../interfaces';
 import 'rxjs/add/operator/merge';
```
Comment (robwormald) [inline src/database/query_observable.ts]: ditto here - `import {merge} from 'rxjs/operator/merge'` , `merge.call()`

**Comment by jeffbcross** [comment]
I'll ditch the side-effect operators and observables.

**Pull request closed without merging**